src/Microsoft.AspNet.Http.Core/Extensions/MapWhenMiddleware.cs

```diff
@@ -20,27 +20,13 @@ public MapWhenMiddleware([NotNull] RequestDelegate next, [NotNull] MapWhenOption
 
         public async Task Invoke([NotNull] HttpContext context)
         {
-            if (_options.Predicate != null)
+            if (_options.Predicate(context))
```

Comment:
The NotNull guard is here:
https://github.com/aspnet/HttpAbstractions/pull/283/files#diff-93bfbc9b10d86e70f618ce3182ccfbeaL27

It used to check before because there were two different options, sync or async.